### DISCUSSION TITLE
SCRUM PREP THREAD MILESTONE 1-1

### DISCUSSION THREAD

**Discussion by Keiven-Soto** [uprm-inso4116-2024-2025-s1/semester-project-multiplayer-dominoes#74] [General]
Hello everyone! This is a space where you can update and communicate the tasks that you will be doing for the week

**Comment by Saul-Caban-UPRM** [comment]
Met with members of the [UI team] in person to discuss art style for assets, agreed that the assets should be pixel art for simplicity and nostalgia purposes. Created a full 28 set of dominoes drawn by me for the task #35  and started to work on the achievement assets for the task #66.  Haven't closed task #35 reason is incase there's need to be changes of colors matching the web pages aesthetics.

**Comment by AngelCIICMorales** [comment]
This week I held a meeting with the group, team leaders and the professor in order to discuss several aspects of the project. I alongside Keiven evaluated pull requests and calculated the total expected score of the milestone.

**Comment by AnthonyMed13** [comment]
Started researching how to do the basic user registration end points which is task #18. Started working on documentation task #19 which are the needs and task #32 which are the domain requirements.

**Comment by Keiven-Soto** [comment]
Calculated expected score for milestone 1. Met with the team and discussed how the outcome of the project should look like for milestone one. Had a meeting with the professor where we talked about the components of the project. I also reviewed some pull requests and held scrum meetings.

**Comment by sandy-marrero** [comment]
Working on Personas tasks, 2 fully developed at the moment. ([Link](https://docs.google.com/document/u/1/d/11d83w8w16uDR34pI4qv3hM5CFbnoj47n0w2totGlc18/edit)) #48 
Working on Derived Goals ([Link](https://docs.google.com/document/d/1eDejBN9PN6dMOEn2-Ccq6UR6YwzqHXzpgPgxTfr_JLE/edit)) #44 
Working on Function Signatures ([Link](https://docs.google.com/document/u/1/d/1VRapilPMtNMFyCzaGecT4h5clO9aTjxfqZwHGstBQYA/edit)) #46

**Comment by PinkSylvie** [comment]
This week I started prepping the questions to QA stakeholders to write the rough sketch for our documentation. I also created the tasks that were missing for the Milestone 1 documentation.

**Comment by JadiellMV** [comment]
Met with the managers, team leaders, and teammates to discuss aspects of the projects and the class, started researching how to do the basic user authentication end points (task #17), and started working on creating a user object (task #20).

**Comment by Joaquin939** [comment]
This week I started researching reliable websites to extract the basic rules of dominoes for the project, focusing on sources that offer clear and accurate game guidelines. The aim is to ensure the project follows standard gameplay rules.

The best ones I could find so far are these links:

https://www.dominorules.com/straight-dominoes
https://www.hasbro.com/common/instruct/dominoesbymiltonbradley.pdf

**Comment by fernandopizarro1** [comment]
Milestone 1 - Week 1
Met with the Gameplay / AI team to discuss strategies. Started implementing the basic AI for the domino game. #11 Implemented the framework for the project. #31

**Comment by ebrain-mendoza121** [comment]
This week started research on implementing a lobby system UI to create or join lobbies that will later be adjusted with the game functions and design

**Comment by print-Eruki** [comment]
Research on what is needed on the database in order to establish four clients in a single websocket and what is needed for extra features (leaderboard). All of this in order to design a ERD and architecture diagram.

**Comment by RicardoJRSs** [comment]
Started research on sign in/login implementation, researching ways to implement lobbies, attended meetings.

**Comment by FrancesC8** [comment]
Milestone 1 - Week 1
I've been mostly working on delivering the initial backend setup and skeleton based on the technologies discussed (Node.js with Express.js and MongoDB). It also includes an initial code structure (Model/Handler/Controller) for organization and readability. Lastly, documentation for how to set it up locally. [PR](https://github.com/uprm-inso4116-2024-2025-s1/semester-project-multiplayer-dominoes/pull/72) is ready for review. 

Next steps is to help troubleshoot any setup and potentially work on defining data models based on ERGs created by others. Task [#28](https://github.com/uprm-inso4116-2024-2025-s1/semester-project-multiplayer-dominoes/issues/28)

**Comment by Adriana-Isabel** [comment]
I finished the wireframe and uploaded it. https://github.com/uprm-inso4116-2024-2025-s1/semester-project-multiplayer-dominoes/issues/36

I should be finished with the Mock-Up Landing page by next Thursday.

I discussed with @Saul-Caban-UPRM the fonts that were talked about for the web app, which will be used for the mock-up landing page. I am open to other suggestions about fonts, color palettes, designs, anything. Please feel free to give ideas! :)

**Comment by DavidCuevas3** [comment]
I started working on the figma design for the mutliplayer leaderboard screen, which will display competitive and rank statistics for players.
#45

**Comment by ianramospr** [comment]
I began working on the Figma design for the dynamic avatar frames, which will change depending on the user's winning streak. #39

**Comment by Negroni1** [comment]
I started researching WebSockets for multiplayer implementation, which includes what are UUIDs and what a communication protocol that is handshake_ based is, and how they are needed for a multiplayer game  #61

**Comment by G4vp** [comment]
Created a class for the table and dominoes logic. The table holds the information of the current game state and can place the selected dominoes in the correct position. 

I am currently working on an experimental version of the game, where you initially get 7 dominoes to choose from to place on the table, and you can draw a domino from the remaining dominoes pool.

**Comment by nsfrontera** [comment]
Met with the team to discuss tasks, Overview the work that needed to be done, researched my tasks and wrote down ideas on how to complete them, and talked with team members to see the order of tasks and progress on started task.

**Comment by ilikecheese117** [comment]
Began working on the save system for the dominoes game and testing it to make sure that the dominoes game is stored whenever players leave the match so that they can come back to it.

**Comment by Gaby-Concepcion** [comment]
Milestone 1 (week 1)

For this first week of milestone one, I prioritized understanding where I could focus my efforts in terms of tasks for both the documentation of the project and the core game features as a member of the _'Gameplay/AI'_ team. These tasks include: 

- Issue #51: "Implementing mouse function and interaction on the GameState."
- Issue #52: "Develop a score system to keep track of the score and assigns it to the correct player."
- Issue #55: "Include and illustrate the Team on the documentation. (1.1)"
- Issue #57: "Include and elaborate on user stories, epics, and features on the documentation. (2.2.1)"

Due to exams this first week, progress on these issues was slow but moving forward my hope is to work through a majority of these during the weekend and make any necessary final tweaks or adjustments early into our second week of the milestone.

**Comment by rukitoyordan** [comment]
**Milestone 1 - Week 1** ♟
During the first week of Milestone 1, I worked on creating a Discord server for the team to coordinate properly before organizing issues and discussions on GitHub. Presented a webhook so that Discord updates developers on GitHub progression if they do not have GitHub's mobile app installed or wish for more organized information flow.

Additionally, I researched on board game asset creation (#33; information will be posted soon) and coordinated with @frances-sepulveda on Avatar Creation through Figma that would be to enhance and be worked on next week in conjunction with #34 and possible #39.

I will find some framework to also start working on contributing to the milestone's documentation requirements.

**Comment by frances-sepulveda** [comment]
During the first week of Milestone 1, I worked with @Saul-Caban-UPRM  and @rukitoyordan  on the hand-drawn creation of avatars for the game. This work is aimed at enhancing our visual assets and will be further developed in conjunction with tasks #34 and #39.

**Comment by Jvstinoo** [comment]
During this week, I worked on initial implementation of #30 (PauseScreen), a screen which includes resume game and quit game buttons. Waiting for gameState and menuState implementations to finish implementing those two buttons (they should redirect to these two states).
